kv/replica_slice_test.go

```diff
@@ -110,3 +117,25 @@ func TestReplicaSetMoveToFront(t *testing.T) {
 		t.Errorf("expected order %s, got %s", exp, stores)
 	}
 }
+
+func TestReplicaSetRandPerm(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	rs := createReplicaSlice()
+	r := rand.New(rand.NewSource(randutil.NewPseudoSeed()))
+	rs.randPerm(0, 4, r.Perm)
+	exp := []roachpb.StoreID{1, 2, 3, 4, 5}
+	if stores := getStores(rs); !reflect.DeepEqual(stores, exp) {
+		t.Errorf("expected order %s, got %s", exp, stores)
+	}
+	rs.randPerm(2, 2, r.Perm)
+	if stores := getStores(rs); !reflect.DeepEqual(stores, exp) {
+		t.Errorf("expected order %s, got %s", exp, stores)
+	}
+
+	rs.randPerm(0, 2, r.Perm)
+	stores := getStores(rs)[3:5]
+	exp = exp[3:5]
+	if !reflect.DeepEqual(stores, exp) {
```

Comment:
as @tamird suggested, get use a known seed and verify that you actually get rearranged replica sets (in particular guarding against one-offs: get an example in which the first and last item of the shuffled part of the slice are not the same).